components/tidb_query/src/rpn_expr/impl_json.rs

```diff
@@ -55,6 +56,83 @@ fn json_unquote(arg: &Option<Json>) -> Result<Option<Bytes>> {
     })
 }
 
+fn json_with_bytes_validator(expr: &tipb::Expr) -> Result<()> {
+    let children = expr.get_children();
+    // min_args = 2
+    // args should be like `&Option<Json> , &[&Option<Bytes>]`.
+    super::function::validate_expr_return_type(&children[0], EvalType::Json)?;
+    for i in 1..children.len() {
+        super::function::validate_expr_return_type(&children[i], EvalType::Bytes)?;
+    }
+    Ok(())
+}
+
+/// args should be like `&Option<Json> , &[&Option<Bytes>]`.
+#[rpn_fn(raw_varg, min_args = 2, extra_validator = json_with_bytes_validator)]
+#[inline]
+fn json_extract(args: &[ScalarValueRef]) -> Result<Option<Json>> {
+    // args should be at least 2
+    let j: &Option<Json> = args[0].as_ref();
+    let j = match j.as_ref() {
+        None => return Ok(None),
+        Some(j) => j.to_owned(),
+    };
+
+    let mut path_expr_list = vec![];
+    for i in 1..args.len() {
+        let json_path: &Option<Bytes> = args[i].as_ref();
+
+        let json_path = match json_path.as_ref() {
+            None => return Ok(None),
+            Some(p) => p.to_owned(),
+        };
+        let json_path = eval_string_and_decode(json_path)?;
+
+        let path_expr = parse_json_path_expr(&json_path)?;
+
+        path_expr_list.push(path_expr);
+    }
+
+    Ok(j.extract(&path_expr_list))
+}
+
+/// args should be like `&Option<Json> , &[&Option<Bytes>]`.
+#[rpn_fn(raw_varg, min_args = 2, extra_validator = json_with_bytes_validator)]
+#[inline]
+fn json_remove(args: &[ScalarValueRef]) -> Result<Option<Json>> {
+    // args should be at least 2
+    let j: &Option<Json> = args[0].as_ref();
+    let mut j = match j.as_ref() {
+        None => return Ok(None),
+        Some(j) => j.to_owned(),
+    };
+
+    let mut path_expr_list = vec![];
+    for i in 1..args.len() {
+        let json_path: &Option<Bytes> = args[i].as_ref();
+
+        let json_path = match json_path.as_ref() {
+            None => return Ok(None),
+            Some(p) => p.to_owned(),
+        };
+        let json_path = eval_string_and_decode(json_path)?;
+
+        let path_expr = parse_json_path_expr(&json_path)?;
+
+        path_expr_list.push(path_expr);
+    }
+
+    j.remove(&path_expr_list)?;
+    Ok(Some(j))
+}
+
+#[inline]
+fn eval_string_and_decode(b: Bytes) -> Result<String> {
```

Comment:
You don't need to port this function (and even if you would, you should port it to some common place instead of here).